### PR TITLE
Add basic tests of dataframe scan

### PR DESCRIPTION
**Pull request by wence-**
## Description
Also assert that unsupported file scan operations raise.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf_polars/tests/test_scan.py [hunk 1]
```diff
@@ -86,3 +87,12 @@ def test_scan(df, columns, mask):
...
     assert_gpu_result_equal(q)
+
+
+def test_scan_unsupported_raises(tmp_path):
+    df = pl.DataFrame({"a": [1, 2, 3]})
+
+    df.write_ndjson(tmp_path / "df.json")
+    q = pl.scan_ndjson(tmp_path / "df.json")
+    with pytest.raises(NotImplementedError):
+        _ = translate_ir(q._ldf.visit())
```
Comment (brandon-b-miller) [inline python/cudf_polars/tests/test_scan.py:98]: It's a little unclear what we're invoking on this line and expecting to fail. Is it the translation of the plan to IR? If this is a pattern we might use in the testing codebase, can we have a utility like `assert_gpu_result_equal` that we use that says what this is trying to do, either in the name or the docs?

**Review by brandon-b-miller — APPROVED** [review]
One question otherwise LGTM

File: python/cudf_polars/tests/test_scan.py (hunk 1 above)
Comment (wence-) [reply python/cudf_polars/tests/test_scan.py:98]: Good idea

Comment (wence-) [reply python/cudf_polars/tests/test_scan.py]: Done.

**Pull request merged**